### ISSUE TITLE
[Question] How to sequence generation of a cluster and provisioning

### ISSUE THREAD

**Issue by jorgemoralespou** [alekc/terraform-provider-kubectl#71]
I'm working on a module that provisions a cluster and then creates kubernetes resources via this plugin. The problem I'm facing is that `terraform plan` fails because this plugin depends on a kubeconfig_file that a previous eks process generates. I've seen #18 but it doesn't fully work for me, as it only works if I do a stepped approach where in one run the cluster is created and in a second run the kubectl_manifests are applied to the cluster, otherwise I get:

```
Error: Invalid for_each argument

  on ../../modules/educates/16-kyverno.tf line 44, in resource "kubectl_manifest" "app_package_installs_kyverno_package":
  44:   for_each  = try(data.kubectl_path_documents.app_package_installs_kyverno_manifests[0].manifests, toset([]))
    ├────────────────
    │ data.kubectl_path_documents.app_package_installs_kyverno_manifests[0].manifests is a map of string, known only after apply 
The "for_each" map includes keys derived from resource attributes that cannot be determined until apply, and so Terraform cannot determine the full set of keys that will identify the instances of this resource.

When working with unknown values in for_each, it's better to define the map keys statically in your configuration and place apply-time results only in the map values.

Alternatively, you could use the -target planning option to first apply only the resources that the for_each value depends on, and then apply a second time to fully converge.
```

I'm guessing that because can't query the cluster for current state.

Is there a way to configure the module so that it doesn't fail and in case where the cluster does not exist the plan just outputs a new creation of the resource?

Otherwise, how is people solving these types of scenarios. I'm finding terraform quite not the best tool for what I'm trying to do, or maybe I'm not doing it the recommended way.

**Comment by jorgemoralespou** [comment]
ok, so it seems that for some reason data_source `kubectl_path_documents` and `kubectl_path_documents` combinations need the cluster to exist (or be queried). If I instead use plain `kubectl_manifest` instead of dynamically loading the files from the filesystem it works fine.

I opened an issue (https://github.com/alekc/terraform-provider-kubectl/issues/63) with this behaviour, but it seems that my analysis there was incorrect, as same manifests applied in a different way work fine.

**Comment by alekc** [comment]
Since you have closed #63 I am going to close this as well. Feel free to reopen if needed

**Comment by jorgemoralespou** [comment]
This is still an issue. When using kubectl_path_manifests with a cluster created via a previous module, final values can't seem to be calculated until after apply, but if I instead use plain kubectl_manifests they do apply fine. There seems to be some difference that prevent me from using kubectl_path_manifests that would like to understand. I will try to do a reproducer project.
